crates/matrix-sdk/src/http_client.rs

```diff
@@ -108,7 +111,8 @@ impl HttpClient {
         request: Request,
         config: Option<RequestConfig>,
         homeserver: String,
-        session: Option<&Session>,
+        access_token: Option<&str>,
+        user_id: Option<&UserId>,
```

Comment:
The signature of this pretty fundamental function is getting longer and longer. I think we should think through ways of making calls to it easier to read.

Reply:
We could have a `SessionConfig` struct (or something) that contains the 4 last parameters

Reply:
Perhaps `RequestOptions`? I'm a bit biased since vodozemac recently got something called `SessionConfig`.

Reply:
I wouldn't know what's in `RequestConfig` vs `RequestOptions` though. Maybe we could have two structs: `ServerConfig` with `homeserver` and `server_versions` and `Credentials` with `user_id` and `access_token`.

Reply:
If we want to swap this whole thing out for something `tower`-based as discussed recently, I'd also be okay with postponing these API questions until we do that.

Reply:
Yes, we could.